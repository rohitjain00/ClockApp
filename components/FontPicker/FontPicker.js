import React from 'react';
import {Picker, StyleSheet, Text, View} from 'react-native';
import PropTypes from 'prop-types';

export function FontPicker(props) {
    function changeValue(text) {
        props.onChange(text);
    }

    function pickerList() {
        return (props.items.map((x, i) => {
            return (<Picker.Item label={x} key={i} value={x}/>);
        }));
    }

    return (
        <View style={styles.container}>
            <Text>{props.text}</Text>
            <Picker
                selectedValue={props.value}
                onValueChange={(value) => {
                    changeValue(value);
                }}>
                {pickerList()}
            </Picker>
        </View>
    );
}

const styles = StyleSheet.create({
    container: {
        marginTop: 20,
        textAlign: 'center',
    },
});

FontPicker.propTypes = {
    text: PropTypes.string,
    value: PropTypes.string,
    items: PropTypes.arrayOf(PropTypes.string),
    onChange: PropTypes.func
}
