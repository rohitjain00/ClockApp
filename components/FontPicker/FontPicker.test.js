//CustomColorPicker.test.js
import React from 'react';
import {render} from '@testing-library/react-native';
import {FontPicker} from "./FontPicker";
import {configure, mount} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({adapter: new Adapter()});

describe('FontPicker Component', () => {
    const mockOnPress = jest.fn();
    const component = <FontPicker onChange={mockOnPress} text={'testing component'} items={['test', 'test2']}
                                  value={'normal'}/>
    let wrapper;
    beforeEach(() => {
        wrapper = mount(component);
        jest.clearAllMocks();
    })
    it('FontPicker Props Validation', async () => {
        const componentProps = wrapper.props();
        expect(componentProps.onChange).toEqual(mockOnPress);
        expect(componentProps.text).toEqual('testing component');
        expect(componentProps.value).toEqual('normal');
        expect(componentProps.items).toEqual(['test', 'test2'])
    });
    it('Font Picker Snapshot Validation', async () => {
        const tree = render(component).toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('Font Picker Action Validation', async () => {
        // https://github.com/enzymejs/enzyme/issues/2263#issuecomment-568585047
        // According to above statement We cannot interact with state inside a function with useState.
    });
    it('Font Picker onChange handler', async () => {
        wrapper.props().onChange('monospace');
        expect(mockOnPress).toHaveBeenCalled();
        expect(mockOnPress).toHaveBeenCalledTimes(1);
        wrapper.props().onChange('normal');
        expect(mockOnPress).toHaveBeenCalledTimes(2);
    });
});