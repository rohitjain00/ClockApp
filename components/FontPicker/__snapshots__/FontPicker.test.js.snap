// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FontPicker Component Font Picker Snapshot Validation 1`] = `
<View
  style={
    Object {
      "marginTop": 20,
      "textAlign": "center",
    }
  }
>
  <Text>
    testing component
  </Text>
  <View>
    <RCTPicker
      items={
        Array [
          Object {
            "label": "test",
            "textColor": undefined,
            "value": "test",
          },
          Object {
            "label": "test2",
            "textColor": undefined,
            "value": "test2",
          },
        ]
      }
      onChange={[Function]}
      selectedIndex={0}
      style={
        Array [
          Object {
            "height": 216,
          },
          undefined,
        ]
      }
    />
  </View>
</View>
`;
